File: algorithms/2_medium/M/01899_merge_triplets_to_form_target_triplet/solution.cpp
class Solution {
public:
    static bool mergeTriplets(const vector<vector<int>>& triplets, const vector<int>& target) {
        bool seen_0 = false;
        bool seen_1 = false;
        bool seen_2 = false;

        for (const auto& t : triplets) {
            if (t[0] <= target[0] && t[1] <= target[1] && t[2] <= target[2]) {
                seen_0 |= (t[0] == target[0]);
                seen_1 |= (t[1] == target[1]);
                seen_2 |= (t[2] == target[2]);
            }

            if (seen_0 && seen_1 && seen_2) {
                return true;
            }
        }

        return false;
    }
};

File: algorithms/2_medium/M/01899_merge_triplets_to_form_target_triplet/solution.rs
impl Solution {
    pub fn merge_triplets(triplets: Vec<Vec<i32>>, target: Vec<i32>) -> bool {
        let (mut seen_0, mut seen_1, mut seen_2) = (false, false, false);

        for t in triplets {
            if t[0] <= target[0] && t[1] <= target[1] && t[2] <= target[2] {
                seen_0 |= (t[0] == target[0]);
                seen_1 |= (t[1] == target[1]);
                seen_2 |= (t[2] == target[2]);
            }

            if seen_0 && seen_1 && seen_2 {
                return true;
            }
        }

        false
    }
}
